### PR TITLE
New: Add selectionStart and selectionEnd

### PR DESCRIPTION
**Pull request by stramel**
- Added `selectionStart` and `selectionEnd` properties to better mimic `textarea` and allow it to really function as a replacement.
- Fixed a few lint errors

@notwaldorf Seems like you have the most commits. Figured you might be able to review.

### CODE REVIEW

**Comment by btopro** [comment]
+1. This allows events of what text is highlighted to bubble up correctly for start and end positions. as @stramel mentioned this would help allow it to be that much closer to a direct replacement for the `textfield` primative

**Comment by stramel** [comment]
@notwaldorf @e111077 Would one of you two mind looking at this for me? :smile:

File: paper-textarea.html [hunk 1]
```diff
@@ -96,18 +96,18 @@
 
     behaviors: [
       Polymer.PaperInputBehavior,
-      Polymer.IronFormElementBehavior
+      Polymer.IronFormElementBehavior,
```
Comment (notwaldorf) [inline paper-textarea.html]: What's up with all these trailing commas?

File: .travis.yml [hunk 2]
```diff
@@ -1,8 +1,5 @@
 language: node_js
-sudo: required
-before_script:
-  - npm install -g polymer-cli
-  - polymer install --variants
+sudo: false
```
Comment (notwaldorf) [inline .travis.yml]: Why is this also changing as part of this PR? We prefer keeping all the `travis.yml` files consistent across all elements, so if one needs to change, they all should, via a `tedium` pass. In any case, it doesn't belong in this PR

File: bower.json [hunk 3]
```diff
@@ -70,7 +70,7 @@
         "iron-validator-behavior": "PolymerElements/iron-validator-behavior#^1.0.0",
         "paper-icon-button": "PolymerElements/paper-icon-button#^1.0.0",
         "test-fixture": "PolymerElements/test-fixture#^1.0.0",
-        "web-component-tester": "^4.0.0",
+        "web-component-tester": "^5.0.0",
```
Comment (notwaldorf) [inline bower.json]: do you _require_ version 5? Nothing in this PR would say you do

File: polymer.json [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (notwaldorf) [inline polymer.json]: Same comment as above

File: paper-textarea.html (hunk 1 above)
Comment (stramel) [inline paper-textarea.html]: https://google.github.io/styleguide/jsguide.html#features-arrays-trailing-comma

Can revert it if you prefer

File: .travis.yml (hunk 2 above)
Comment (stramel) [inline .travis.yml]: I will revert and bring up for a tedium pass instead.

File: bower.json (hunk 3 above)
Comment (stramel) [inline bower.json]: Forgot to revert, I do this in testing so I can use `.only`

File: polymer.json (hunk 4 above)
Comment (stramel) [inline polymer.json]: Will remove.

File: .travis.yml (hunk 2 above)
Comment (stramel) [inline .travis.yml]: https://github.com/PolymerLabs/tedium/issues/57

**Comment by stramel** [comment]
@notwaldorf I made all changes requested with a comment about the trailing commas and a PR in Tedium to handle the additional updates. Let me know if there is anything else I need to do

**Comment by stramel** [comment]
Ping @notwaldorf

**Comment by stramel** [comment]
@notwaldorf Can you PTAL please?

**Comment by notwaldorf** [comment]
Thanks! Sorry for the delay :/

**Comment by stramel** [comment]
@notwaldorf No worries! Thank you for the feedback and getting it merged!
